addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/protocol/HyperionConnection.java

```diff
@@ -0,0 +1,153 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.hyperion.internal.protocol;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openhab.binding.hyperion.internal.protocol.effect.Effect;
+import org.openhab.binding.hyperion.internal.protocol.request.ClearAllCommand;
+import org.openhab.binding.hyperion.internal.protocol.request.ClearCommand;
+import org.openhab.binding.hyperion.internal.protocol.request.ColorCommand;
+import org.openhab.binding.hyperion.internal.protocol.request.EffectCommand;
+import org.openhab.binding.hyperion.internal.protocol.request.HyperionCommand;
+import org.openhab.binding.hyperion.internal.protocol.request.ServerInfoCommand;
+import org.openhab.binding.hyperion.internal.protocol.request.TransformCommand;
+import org.openhab.binding.hyperion.internal.protocol.transform.ValueGain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.jayway.jsonpath.JsonPath;
+
+/**
+ * The {@link HyperionConnection} class is responsible for handling
+ * the communication with the Hyperion server.
+ *
+ * @author Daniel Walters - Initial contribution
+ */
+public class HyperionConnection {
+
+    private Logger logger = LoggerFactory.getLogger(HyperionConnection.class);
+    private InetAddress address;
+    private int port;
+    private List<HyperionStateListener> listeners = new ArrayList<HyperionStateListener>();
+
+    private double valueGain = Double.MIN_VALUE;
+
+    public final static String PROPERTY_VALUEGAIN = "valueGain";
+    private final static String JSONPATH_VALUEGAIN = "$.info.transform[0].valueGain";
+
+    public HyperionConnection(InetAddress address, int port) {
+        this.setAddress(address);
+        this.port = port;
+    }
+
+    public HyperionConnection(String sAddress, int port) throws UnknownHostException {
+        this(InetAddress.getByName(sAddress), port);
+    }
+
+    public void addListener(HyperionStateListener listener) {
+        if (!listeners.contains(listener)) {
+            listeners.add(listener);
+        }
+    }
+
+    public void removeListener(HyperionStateListener listener) {
+        if (listeners.contains(listener)) {
+            listeners.remove(listener);
+        }
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public InetAddress getAddress() {
+        return address;
+    }
+
+    public void setAddress(InetAddress address) {
+        this.address = address;
+    }
+
+    public String send(HyperionCommand command) throws IOException {
+        Gson gson = new Gson();
```

Comment:
This is potentially expensive, so you should imho use a field and reuse the GSON instance.